src/renderer/actions/queries.ts

```diff
@@ -102,18 +110,19 @@ function shouldUpdateQuery(query, selectedQuery, state) {
   return true;
 }
 
-export function appendQuery(query) {
+export function appendQuery(query: string): ThunkResult<void> {
   return (dispatch, getState) => {
-    const currentQuery = getCurrentQuery(getState()).query;
+    const queryState = getCurrentQuery(getState());
+    const currentQuery = queryState.query;
     const newLine = !currentQuery ? '' : '\n';
     const appendedQuery = `${currentQuery}${newLine}${query}`;
-    if (!currentQuery.isExecuting) {
+    if (!queryState.isExecuting) {
```

Comment:
Issue found: It was accessing `isExecuting` from the query text, which does not existing since it is a string, instead of the query state.